opm/autodiff/ParallelOverlappingILU0.hpp

```diff
@@ -442,17 +442,30 @@ namespace Opm
         upper.resize( A.N() );
         inv.resize( A.N() );
 
-        lower.reserveAdditional( 2*A.N() );
+        // Count the lower and upper matrix entries.
+        size_type numLower = 0;
+        size_type numUpper = 0;
+        const auto endi = A.end();
+        for (auto i=A.begin(); i!=endi; ++i) {
+          const size_type iIndex = i.index();
+          for (auto j = (*i).begin(); j.index() < iIndex; ++j) {
```

Comment:
I'm fine with this being merged once the counting of numLower and numUpper is done as follows: 
```c++
       // Count the lower and upper matrix entries.
        size_type numLower = 0;
        size_type numUpper = 0;
        const auto endi = A.end();
        for (auto i=A.begin(); i!=endi; ++i) {
          const size_type iIndex = i.index();
          size_type numLowerRow = 0;
          for (auto j = (*i).begin(); j.index() < iIndex; ++j) {
              ++numLowerRow;
          }
          numLower += numLowerRow;
          numUpper += (*i).size() - numLowerRow - 1 ;
        }
        assert( numLower + numUpper + A.N() == A.nonzeroes() );
              
        lower.resize( numLower );
        upper.resize( numUpper );
        inv.resize( A.N() );
```